# Laptop Analysis

## Load and Inspect the Dataset

First, I will load the dataset and check its basic structure.

In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('./data/laptops.csv')
df

,Title,Price,Image URL,Specifications
0,"Laptop HP OmniBook 3 16-by0028nwx 16"" R5 40 8G...",1 999.00 zł,https://f00.esfr.pl/foto/4/192315626929/e8ea62...,"{'Ekran': '16 "", 1920 x 1200 pikseli', 'Proce..."
1,"Laptop gamingowy Lenovo LOQ 15IAX9 15,6"" 144Hz...",3 499.00 zł,https://f00.esfr.pl/foto/7/177142838209/75780b...,"{'Ekran': '15,6 "", 1920 x 1080 pikseli 144 Hz..."
2,Laptop gamingowy Dell Alienware 16 Aurora AC16...,3 299.00 zł,https://f00.esfr.pl/foto/2/188164628089/f7bd6d...,"{'Ekran': '16 "", 2560 x 1600 pikseli 120 Hz',..."
3,Laptop gamingowy ASUS TUF Gaming F16 FX607VJB-...,3 299.00 zł,https://f00.esfr.pl/foto/4/191218243201/6365f8...,"{'Ekran': '16 "", 1920 x 1200 pikseli 144 Hz',..."
4,"Laptop Lenovo IdeaPad Slim 3 16ARP10 16"" R5 75...",2 699.00 zł,https://f00.esfr.pl/foto/4/187619326041/eed87c...,"{'Ekran': '16 "", 1920 x 1200 pikseli 60 Hz', ..."
...,...,...,...,...
1070,"Laptop biznesowy HP 470 G9 17,3"" i7-1255U 16GB...",NaN,https://f00.esfr.pl/foto/9/120043513681/2a841f...,"{'Ekran': '17,3 "", 1920 x 1080 pikseli', 'Pro..."
1071,Laptop ASUS Vivobook 16 Flip TP3607SH-RJ007W O...,NaN,https://f00.esfr.pl/foto/7/172240043369/f2f3fe...,"{'Ekran': '16 "", ekran dotykowy 2880 x 1800 pi..."
1072,Laptop biznesowy ASUS ExpertBook B9 OLED B9403...,NaN,https://f00.esfr.pl/foto/8/184604732593/2dc278...,"{'Ekran': '14 "", 2880 x 1800 pikseli 120 Hz',..."
1073,Laptop gamingowy HP OMEN 16-u0042nw 16'' 240Hz...,NaN,https://f00.esfr.pl/foto/2/137623950793/656345...,"{'Ekran': '16 "", 2560 x 1600 pikseli 240 Hz',..."


### Dataset Shape

Let's check the number of rows and columns.

In [3]:
df.shape

(1075, 4)

### Column Names

Let's check the column names.

In [4]:
df.columns

Index(['Title', 'Price', 'Image URL', 'Specifications'], dtype='object')

### Data Types

Let's check the data types and the structure of the dataset.

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1075 entries, 0 to 1074
Data columns (total 4 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   Title           1075 non-null   object
 1   Price           842 non-null    object
 2   Image URL       1075 non-null   object
 3   Specifications  1075 non-null   object
dtypes: object(4)
memory usage: 33.7+ KB


### Missing Values

Let's check for missing values in each column.

In [6]:
df.isna().sum()

Title               0
Price             233
Image URL           0
Specifications      0
dtype: int64

### Duplicate Rows

Let's check for duplicate rows.

In [7]:
df.duplicated().sum()

0

### Basic Statistics

Let's look at some basic statistics.

In [8]:
df.describe(include="all").T

,count,unique,top,freq
Title,1075,1057,"Laptop biznesowy Dell Pro 16 Plus PB16250 16"" ...",3
Price,842,265,2 999.00 zł,26
Image URL,1075,1075,https://f00.esfr.pl/foto/4/192315626929/e8ea62...,1
Specifications,1075,914,"{'Ekran': '15,3"", 2880 x 1864 pikseli', 'Proc...",8


### Initial Observations

The dataset contains 1,075 laptop records and 4 columns.

There are 233 missing values in the `Price` column. The other columns do not contain missing values.

There are no fully duplicated rows.

The `Price` and `Specifications` columns need further processing before the analysis.